packages/gafl-webapp-service/src/pages/summary/contact-summary/route.js

```diff
@@ -19,186 +19,198 @@ import {
 
-export const checkNavigation = (status, permission) => {
-  if (!permission.isRenewal) {
-    if (!status[ADDRESS_ENTRY.page] && !status[ADDRESS_SELECT.page]) {
-      throw new GetDataRedirect(ADDRESS_LOOKUP.uri)
-    }
+const CONTACT_TEXT_DEFAULT = {
+  EMAIL: 'contact_summary_email',
+  TEXT: 'contact_summary_text_sngl',
+  DEFAULT: 'contact_summary_license_default'
+}
 
-    if (!status[CONTACT.page]) {
-      throw new GetDataRedirect(CONTACT.uri)
-    }
-  }
+const CONTACT_TEXT_NON_PHYSICAL = {
+  EMAIL: 'contact_summary_email',
+  TEXT: 'contact_summary_text_plrl',
+  DEFAULT: 'contact_summary_license_non_physical'
+}
 
-  if (isPhysical(permission)) {
-    if (!status[LICENCE_FULFILMENT.page]) {
-      throw new GetDataRedirect(LICENCE_FULFILMENT.uri)
-    }
-    if (!status[LICENCE_CONFIRMATION_METHOD.page]) {
-      throw new GetDataRedirect(LICENCE_CONFIRMATION_METHOD.uri)
-    }
-  }
+const CONTACT_TEXT_PHYSICAL = {
+  EMAIL: 'contact_summary_email',
+  TEXT: 'contact_summary_text_plrl',
+  DEFAULT: 'contact_summary_license_physical'
 }
 
-const getData = async request => {
-  const status = await request.cache().helpers.status.getCurrentPermission()
-  const permission = await request.cache().helpers.transaction.getCurrentPermission()
+const CHANGE_CONTACT = 'change-contact'
 
-  checkNavigation(status, permission)
+class RowGenerator {
+  constructor (request, permission) {
+    this.request = request
+    this.permission = permission
+    this.labels = request.i18n.getCatalog()
+  }
 
-  status.fromSummary = CONTACT_SUMMARY_SEEN
-  await request.cache().helpers.status.setCurrentPermission(status)
-  const countryName = await countries.nameFromCode(permission.licensee.countryCode)
+  _getContactText (contactTextSpec) {
+    switch (this.permission.licensee.preferredMethodOfReminder) {
+      case HOW_CONTACTED.email:
+        return `${this.labels[contactTextSpec.EMAIL]}${this.permission.licensee.email}`
+      case HOW_CONTACTED.text:
+        return `${this.labels[contactTextSpec.TEXT]}${this.permission.licensee.mobilePhone}`
+      default:
+        return this.labels[contactTextSpec.DEFAULT]
+    }
+  }
 
-  return {
-    summaryTable: getLicenseeDetailsSummaryRows(permission, countryName, request),
-    uri: {
-      licenceSummary: LICENCE_SUMMARY.uri
+  _generateRow (label, text, rawHref, visuallyHiddenText, id) {
+    const href = addLanguageCodeToUri(this.request, rawHref)
+    return {
+      key: {
+        text: label
+      },
+      value: {
+        text
+      },
+      ...(href && {
+        actions: {
+          items: [
+            {
+              href,
+              visuallyHiddenText,
+              text: this.labels.contact_summary_change,
+              attributes: { id }
+            }
+          ]
+        }
+      })
     }
   }
-}
 
-export default pageRoute(CONTACT_SUMMARY.page, CONTACT_SUMMARY.uri, null, nextPage, getData)
+  generateRow (label, text, rawHref, visuallyHiddenText, id) {
```

Comment:
could this be renamed in similar way as below? ( generateAddressRow etc)

Reply:
Not a bad point, it had crossed my mind - perhaps generateStandardRow?

Reply:
yeah that would work